service/history/workflow/context.go

```diff
@@ -919,13 +919,26 @@ func (c *ContextImpl) ReapplyEvents(
 
 // TODO: remove `ms` parameter again (added since it's not possible to initialize a new Context with a specific MutableState)
 func (c *ContextImpl) UpdateRegistry(ctx context.Context, ms MutableState) update.Registry {
+	var currentVersion int64
+	if c.MutableState != nil {
+		currentVersion = c.MutableState.GetCurrentVersion()
+	} else {
+		currentVersion = ms.GetCurrentVersion()
+	}
```

Comment:
I'll trust you on this logic! Hopefully one day we can reduce the number of references to MS...